modules/spring/src/test/java/org/apache/ignite/internal/processors/resource/GridSpringResourceInjectionSelfTest.java

```diff
@@ -304,6 +323,47 @@ private void setDummyResourceBean(AnotherDummyResourceBean dummyRsrcBean) {
             ".GridSpringResourceInjectionSelfTest$AnotherDummyResourceBean' available");
     }
 
+
+    /** */
+    @Test
+    public void testClosureMethodWithDuplicatingResourceClass() {
+        assertError(new IgniteCallable<Object>() {
+            private DuplicatedResourceBean rsrcBean;
+
+            @SpringResource(resourceClass = DuplicatedResourceBean.class, required = false)
+            private void setDummyResourceBean(DuplicatedResourceBean rsrcBean) {
+                this.rsrcBean = rsrcBean;
+            }
+
+            @Override public Object call() {
+                fail();
+
+                return null;
+            }
+        }, grid, NoUniqueBeanDefinitionException.class, "No qualifying bean of type" +
+            " 'org.apache.ignite.internal.processors.resource.GridSpringResourceInjectionSelfTest$DuplicatingResourceBean'" +
```

Comment:
The test is failing at the moment.

suggestion

```
        }, grid, NoUniqueBeanDefinitionException.class, "No qualifying bean of type '" +
            DuplicatedResourceBean.class.getName() + "' available: expected single matching bean but found 2");
```

Reply:
Done. Thanks a lot!